src/bootstrap/utils.ts

```diff
@@ -162,8 +156,9 @@ async function bootstrapState({
     const nodeConnectionManager = new NodeConnectionManager({
       keyRing,
       nodeGraph,
-      proxy,
-      taskManager,
+      quicClientConfig: {} as any, // No connections are attempted
+      crypto: {} as any, // No connections are attempted
+      quicSocket: {} as any, // No connections are attempted
```

Comment:
I think the `NodeManager` wants the `NodeConnectionManager`. It I didn't look to deep into it since any further changes to bootstrapping will be out of scope for this PR.  I think we can remove the NCM as a dependency but as things stand, to avoid a lot of extra dependencies we'd have to fake stuff we know wont be used. 

As for why the `QUICSocket` is required by `NodeConnectionManager`, we have to share the socket between the server and client connections to do hole punching.

Reply:
Ok I guess this usage is just a bit weird for me, it's basically an escape hatch, no different from using `undefined as any` and passing it into.

This can cause errors in the future if we are not careful and end up calling/using those objects.

It's better we clearly distinguish bootstrapping code-paths so that these variables can be optional, or have an entirely different set of functions that is called (a clear distinct codepath) during bootstrapping.

I imagine that because it's the equivalent of agent start and then shutdown, we are just telling it to optionally not actually start certain network functionality that isn't necessary during bootstrapping.

I'm thinking that this should imply a separate static creation function if so.

Can you create an issue to track this problem because this could be a programming "mine" that might we might step on in the future.

Reply:
Created a new issue at #539.